### PR TITLE
Feat: component countdown dynamic v2

### LINKED ISSUE #75: Component: Countdown (dynamic)
Image

https://github.com/47ng/nuqs/blob/295fae1ea745f481cbf00dd979157d6f75bb9974/packages/docs/src/components/countdown.tsx

### PR DESCRIPTION
**Pull request by IounOm**
## What does it do?
- edit countdown component in PR #115 

## Screenshot
Screenshot 2567-10-09 at 16 47 52
Screenshot 2567-10-09 at 16 48 08

## Related issue
- close #75

### CODE REVIEW

**Comment by JeanneGrenet** [comment]
When I'm trying to test your branch I have this error in my terminal 

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.` 

Did you check the errors before asking for review ?

**Comment by IounOm** [comment]
The component uses `useEffect` and `MotionNumber` to render the countdown time animation, which must run exclusively on the client-side. When this component is used or rendered on the server-side, it triggers a warning. The solution is to declare the props using `client:load `or `client:only="react" `in the `Countdown` component, which functions as an interactive island component. You can read more details at [Astro Islands Documentation](https://docs.astro.build/en/concepts/islands/).

**Comment by houssembaltii** [comment]
> I have checked and tested everything before requesting the review. The Countdown component is functioning as expected.
> 
> The component uses `useEffect` and `MotionNumber` to render the countdown time animation, which must run exclusively on the client-side. When this component is used or rendered on the server-side, it triggers a warning. The solution is to declare the props using `client:load `or `client:only="react" `in the `Countdown` component, which functions as an interactive island component. You can read more details at [Astro Islands Documentation](https://docs.astro.build/en/concepts/islands/).
> 
> Screenshot 2567-10-16 at 14 59 14

`client:load` will render the component at server first then rehydrates  when js is loaded, and motion-number will be using useLayoutEffect which it describes in warning , `useLayoutEffect should only be used in components that render exclusively on the client`  . so we need to make this component only renders on client side by using `client:only="react"`

**Comment by IounOm** [comment]
> `client:load` will render the component at server first then rehydrates when js is loaded, and motion-number will be using useLayoutEffect which it causes the warning , `useLayoutEffect should only be used in components that render exclusively on the client` . so we need to make this component only renders on client side by using `client:only="react"`

Okay, I will modify the component declaration in the `countdown.mdx` file to use `client:only="react"`

File: src/components/countdown.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <MotionNumber
+          id={unit}
+          value={value}
+          format={{ minimumIntegerDigits: 2 }}
+          className="font-heading text-2xl"
+        />
+        <p className="font-sans text-xs">{unit}</p>
+      </div>
+    );
+  };
```
Comment (JeanneGrenet) [inline src/components/countdown.tsx:42]: I think it will be easier to maintain if you put this component outside the main component (on the same file)

Comment (IounOm) [reply src/components/countdown.tsx:42]: Ok, I will separate it into another component.

Comment (IounOm) [reply src/components/countdown.tsx]: Updated
- separate TimerDisplay into another component

Screenshot 2567-10-17 at 18 55 10

File: src/content/components/countdown.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <TimerDisplay value="10" unit="Hours" />
+    </div>
+    <div className="flex items-center gap-2">
+      <p>Unit Minutes:</p>
+      <TimerDisplay value="10" unit="Minutes" />
+    </div>
+    <div className="flex items-center gap-2">
+      <p>Unit Seconds:</p>
+      <TimerDisplay value="10" unit="Seconds" />
+    </div>
```
Comment (houssembaltii) [inline src/content/components/countdown.mdx:33]: don't forget to add `client:only="react"`

Comment (IounOm) [reply src/content/components/countdown.mdx]: fixed!

**Pull request merged**